megatron/model/transformer.py

```diff
@@ -1222,11 +1286,106 @@ def set_input_tensor(self, input_tensor):
         forward_step_func"""
         self.input_tensor = input_tensor
 
+    def _build_alibi_tensor(self, tensor, max_seq_len, num_attention_heads):
```

Comment:
Placing this func together with `alibi_mask_func`

Reply:
This func requires an internal variable first_run, so it cannot be placed in the utils.